### PR TITLE
refactor: add TableInfoKey and TableRegionKey

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add TableInfoKey and TableRegion, prepare to split TableGlobalKey.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1728

### CODE REVIEW

File: src/common/meta/src/key.rs [hunk 1]
```diff
@@ -12,16 +12,102 @@
...
+use table_region::{TableRegionManager, TableRegionValue};
+
+use crate::error::{InvalidTableMetadataSnafu, Result, SerdeJsonSnafu};
 pub use crate::key::table_route::{TableRouteKey, TABLE_ROUTE_PREFIX};
+use crate::kv_backend::KvBackendRef;
 
 pub const REMOVED_PREFIX: &str = "__removed";
 
+const TABLE_INFO_KEY_PREFIX: &str = "__table_info";
+const TABLE_REGION_KEY_PREFIX: &str = "__table_region";
```
Comment (Fengys123) [inline src/common/meta/src/key.rs:51]: Why not "__tr" "__ti"?　🤣

Comment (MichaelScofield) [reply src/common/meta/src/key.rs:51]: Because it's the table id that follows these prefix, "__table_info/123" is more clear than "__ti/123".

**Comment by MichaelScofield** [comment]
@Fengys123 PTAL

File: src/common/meta/src/key.rs (hunk 1 above)
Comment (Fengys123) [reply src/common/meta/src/key.rs:51]: "__ti" Indeed is a bit unclear.

**Review by Fengys123 — APPROVED** [review]
LGTM

**Review by WenyXu — APPROVED** [review]

**Pull request merged**